### ISSUE TITLE
Support the CACHEDIR.TAG standard for dist and dist-newstyle

### ISSUE THREAD

**Issue by Profpatsch** [haskell/cabal#9529]
Cache directories tend to be a nuisance for automatic backup tools, because they are very big, which steals network bandwidth and storage capacity from backups.

Users could manually add any Haskell project to their backup CLI with an absolute path, or they could add `--exclude **/dist-newstyle` and pray that no important information on their system uses the same directory name.

Or cabal could implement https://bford.info/cachedir/ and add a `CACHEDIR.TAG` file to its cache directories, meaning tools like `tar --exclude-caches` will ignore the directory. 

Previous implementors of this spec I can see on my system (`find . -iname "cachedir.tag"`):
* Rust’s `cargo` has been doing this for some time.
* mypy does it
* torbrowser
* fontconfig

Basically every cache directory that’s not in `/var/cache` or under `$XDG_CACHDIR` should have this file.

**Comment by Mikolaj** [comment]
I sounds like a great idea to me. Possibly also important to industrial users (with limited resources). Does anybody see any problem with that or can we just mark it `pr-welcome`?

**Comment by ulysses4ever** [comment]
Sounds like a great idea to me!
